cli/azd/cmd/provision.go

```diff
@@ -185,6 +193,14 @@ func (p *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error
 		}, nil
 	}
 
+	if deployResult.SkippedReason == provisioning.DeploymentStateSkipped {
+		return &actions.ActionResult{
+			Message: &actions.ResultMessage{
+				Header: "There are no changes to provision for your application.",
```

Comment:
I added a tag:

<img width="366" alt="image" src="https://github.com/Azure/azure-dev/assets/2163001/96a4cd96-48b6-4824-aad6-fbb57b337579">


But still got this message:
<img width="586" alt="image" src="https://github.com/Azure/azure-dev/assets/2163001/33d99622-5cc4-48b9-a3c8-28e52c2a17e1">


The tag was applied:
<img width="353" alt="image" src="https://github.com/Azure/azure-dev/assets/2163001/c0a7b9b8-f53a-4ef7-bcba-b2050bbfa18f">

Probably need to update logic that detects an infra change.